Zotero/Scenes/Main/Views/MainViewController.swift

```diff
@@ -75,8 +78,10 @@ final class MainViewController: UISplitViewController {
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-        self.set(userActivity: .mainActivity)
-        self.didAppear = true
+        guard let detailCoordinator else { return }
+        let userActivity: NSUserActivity = .mainActivity
```

Comment:
You could maybe create `NSUserActivity` extension, something like: `func create(activity: NSUserActivity, title: String) -> NSUserActivity` so that we can just do `set(userActivity: .create(activity: .mainActivity, title: detailCoordinator.collection.name)` instead of 3 lines every time. But also we only have this in 2 places now, it's fine as it is too. So up to you :).

Reply:
Check https://github.com/zotero/zotero-ios/pull/912/commits/e08d4039b87fe839bd332797a73dc22687c92343